### PR TITLE
fix: disable toolbar when no selection is available

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-4435]

### Description

By default when the Lexical editor starts with the selection `null`, as soon as we enter the editable area, the selection is set. When we focus the editor for the first time by clicking in the toolbar area, the editor still does not have any selection, so clicking triggers no action as the editor does not know to what position to insert new nodes.

I fixed the issue by disabling the toolbar buttons until the selection is selected.

As mentioned, this edge case is only happening for the first focus of the editor.

### How to test

- [Temploy](https://picasso.toptal.net/fx-4435-rte-image-button)

1. check any story of RichTextEditor
2. click in the toolbar area of the editor
  - The toolbar should still be disabled
  - editor should be in focus state
3. click in the editable area
  - the toolbar should not be disabled anymore
4. click outside of the editor to lose focus
  - the toolbar should be disabled
5. click in the toolbar area again
  - the toolbar should not be disabled as the selection is saved inside inner state of lexical editor
  - try to use any button of toolbar and it should focus the editable area and apply expected format/block

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| https://github.com/toptal/picasso/assets/6830426/6dabda2c-801d-4c26-bbf8-a9cc02f90552 | https://github.com/toptal/picasso/assets/6830426/3832226d-d2de-4f00-ae13-a33c14a71e47 |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
8. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
9. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4435]: https://toptal-core.atlassian.net/browse/FX-4435?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mkrl — APPROVED** [review]
Great job! Manually tested, code looks good.

File: packages/picasso-rich-text-editor/src/plugins/api.tsx [hunk 1]
```diff
@@ -72,8 +72,27 @@ export const RTEPluginContextProvider = ({
 }: ToolbarPortalProviderProps) => {
   const [disabledFormatting, setDisabledFormatting] = useState(false)
 
+  /**
+   * When the editor renders, the selection is null.
+   * As soon as we focus the lexical editor (we can see the blinking "I"), the selection is not null.
```
Comment (sashuk) [inline packages/picasso-rich-text-editor/src/plugins/api.tsx:77]: NIT: I was trying to understand why button for italic format needs to blink at first, maybe to use _we can see the blinking text edit cursor_  (this is subjective, so NIT)?

The _text edit cursor_ seems to have quite common meaning

Screenshot 2023-11-03 at 11 18 07

**Review by sashuk — APPROVED** [review]
Awesome description, code looks great, tested in temploy!

File: packages/picasso-rich-text-editor/src/plugins/api.tsx (hunk 1 above)
Comment (sashuk) [reply packages/picasso-rich-text-editor/src/plugins/api.tsx:77]: @TomasSlama Thanks!
